### ISSUE TITLE
I am experiencing 0% GPU utilization when using the NVIDIA RTX 3090 GPU for feature extraction.

### ISSUE THREAD

**Issue by Liuyifan1016** [mahmoodlab/CLAM#196]
When using extract_features_fp, I'm experiencing significantly slow performance, and upon checking the GPU utilization with nvidia-smi, it shows 0% usage. How can I resolve this issue?

**Comment by scjjb** [comment]
@Liuyifan1016 It's likely that you have other resource constraints meaning that your GPU can barely get going. There is a huge amount of data being loaded, and if your input/output speed isn't very very fast then your GPU will just be sat waiting.

**Comment by Liuyifan1016** [comment]
> @Liuyifan1016 It's likely that you have other resource constraints meaning that your GPU can barely get going. There is a huge amount of data being loaded, and if your input/output speed isn't very very fast then your GPU will just be sat waiting.

Thank you for your response.  I have resolved the issue.  GPUs of the 30 series and above require CUDA 11.0 or higher.  The problem was caused by installing a lower version of CUDA as specified in the configuration file, which resulted in the inability to utilize the GPU.  The solution was to update CUDA to a version compatible with the 30 series GPUs or use a GPU series that supports CUDA 10.
